torch/csrc/jit/codegen/cuda/lower_validation.cpp

```diff
@@ -878,6 +884,81 @@ void validateMmaTensors(MmaOp* mma) {
   validate_operand_ids(mma->inB()->as<TensorView>());
 }
 
+//! Note and TODO:
+//!   Currently relying on ldmatrix to
+//!     obtain the correct data layout for turing/ampere
+//!     mma's.
+//!   This restriction will eventually not
+//!    be necessary once the scatter swizzle is ready.
+void validateTuringInput(TensorView* tv) {
+  // Currently only allowed input paths are:
+  //  ldmatrix -> mma or
+  //  ldmatrix -> broadcast -> mma
+  // We actually wouldn't want too much flexibility here since
+  //  this path is very perf critical. But the check itself
+  //  can be made cleaner once we have the correct swizzle
+  //  labeling.
+  // The most generic support would involve build out to
+  //  support any pointwise ops that does not change the
+  //  datalayout.
```

Comment:
Yes, if we use cp async then the prolog ops has to be after ldmatrix because the data go straight from gmem into smem and there's no place to fuse. 

Broadcast discussion in the next thread.